modules/playlists/src/main/java/org/opencastproject/playlists/persistence/PlaylistDatabaseServiceImpl.java

```diff
@@ -0,0 +1,238 @@
+    db = dbSessionFactory.createSession(emf);
+  }
+
+  /**
+   * {@inheritDoc}
+   * @see PlaylistDatabaseService#getPlaylist(String)
+   */
+  @Override
+  public Playlist getPlaylist(String playlistId) throws NotFoundException, PlaylistDatabaseException {
+    return getPlaylist(playlistId, securityService.getOrganization().getId());
+  }
+
+  /**
+   * {@inheritDoc}
+   * @see PlaylistDatabaseService#getPlaylist(String, String)
+   */
+  @Override
+  public Playlist getPlaylist(String playlistId, String orgId) throws NotFoundException, PlaylistDatabaseException {
+    try {
+      return db.execTxChecked(em -> {
+        Optional<Playlist> playlist = getPlaylistById(playlistId, orgId).apply(em);
+        if (playlist.isEmpty()) {
+          throw new NotFoundException("No playlist with id=" + playlistId + " exists");
+        }
+        return playlist.get();
+      });
+    } catch (NotFoundException e) {
+      throw e;
+    } catch (Exception e) {
+      logger.error("Could not retrieve playlist with ID '{}'", playlistId, e);
+      throw new PlaylistDatabaseException(e);
+    }
+  }
+
+  /**
+   * {@inheritDoc}
+   * @see PlaylistDatabaseService#getPlaylists(int, int, SortCriterion)
+   */
+  @Override
+  public List<Playlist> getPlaylists(int limit, int offset, SortCriterion sortCriterion)
+          throws PlaylistDatabaseException {
+
+    try {
+      return db.exec(em -> {
+        CriteriaBuilder criteriaBuilder = em.getCriteriaBuilder();
+        CriteriaQuery<Playlist> criteriaQuery = criteriaBuilder.createQuery(Playlist.class);
+        Root<Playlist> from = criteriaQuery.from(Playlist.class);
+        CriteriaQuery<Playlist> select = criteriaQuery.select(from)
+            .where(criteriaBuilder.isNull(from.get("deletionDate")));
+
+        if (sortCriterion.getOrder().equals(SortCriterion.Order.Ascending)) {
+          criteriaQuery.orderBy(criteriaBuilder.asc(from.get(sortCriterion.getFieldName())));
+        } else if (sortCriterion.getOrder().equals(SortCriterion.Order.Descending)) {
+          criteriaQuery.orderBy(criteriaBuilder.desc(from.get(sortCriterion.getFieldName())));
+        }
+
+        TypedQuery<Playlist> allQuery = em.createQuery(select);
+
+        allQuery.setMaxResults(limit);
+        allQuery.setFirstResult(offset);
+
+        return allQuery.getResultList();
+      });
+    } catch (Exception e) {
+      throw new PlaylistDatabaseException("Error fetching playlists from database", e);
+    }
+  }
+
+  /**
+   * {@inheritDoc}
+   * @see PlaylistDatabaseService#updatePlaylist(Playlist, String)
+   */
+  @Override
+  public Playlist updatePlaylist(Playlist playlist, String orgId) throws PlaylistDatabaseException {
+    try {
+      return db.execTx(em -> {
+        Optional<Playlist> fromDb = getPlaylistById(playlist.getId(), orgId).apply(em);
+        playlist.setUpdated(new Date());
+        if (fromDb.isEmpty()) {
+          em.persist(playlist);
+        } else {
+          em.merge(playlist);
```

Comment:
Merge returns the managed instance:
```suggestion
          return em.merge(playlist);
```